modules/apps/wysiwyg/wysiwyg-web/src/main/java/com/liferay/wysiwyg/web/internal/upgrade/v1_0_0/UpgradeContentPortlet.java

```diff
@@ -0,0 +1,342 @@
+import com.liferay.portal.kernel.xml.Element;
+import com.liferay.portal.kernel.xml.SAXReaderUtil;
+import com.liferay.portal.util.PortalInstances;
+import com.liferay.wysiwyg.web.internal.constants.WysiwygConstants;
+
+import java.sql.PreparedStatement;
+import java.sql.ResultSet;
+import java.sql.SQLException;
+
+import java.util.Collections;
+import java.util.List;
+import java.util.Locale;
+import java.util.Map;
+
+/**
+ * @author Peter Fellwock
+ */
+public class UpgradeContentPortlet extends UpgradeProcess {
+
+	@Override
+	protected void doUpgrade() throws Exception {
+		try {
+			long companyId = PortalInstances.getDefaultCompanyId();
+			
+			Company company = CompanyLocalServiceUtil.getCompany(companyId);
+
+			DDMStructure ddmStructure =
+				DDMStructureLocalServiceUtil.getStructure(
+					company.getGroupId(),
+					PortalUtil.getClassNameId(JournalArticle.class),
+					WysiwygConstants.STRUCTURE_KEY);
+
+			
+			_migrate(ddmStructure.getUserId());
+		}
+		catch (SQLException sqle) {
+			_log.error(
+				"Unable to get PortletPreferences for WYSIWYG Portlet ", sqle);
+		}
+		catch(PortalException pe){
+			_log.error(
+				"Upgrade of WYSIWYG data and portlets not executed " +
+					"DDMStructure/Template not in place", pe);
+		}
+		
+	}
+
+	private JournalArticle _convertWysiwygContent(
+			long userId, long companyId, long groupId, String content)
+		throws Exception {
+
+		ServiceContext serviceContext = new ServiceContext();
+		
+		serviceContext.setAddGroupPermissions(true);
+		serviceContext.setAddGuestPermissions(true);
+		
+		serviceContext.setScopeGroupId(groupId);
+
+		long journalFolderId = _getFolderId(userId, groupId, serviceContext);
+
+		String xmlContent = _getContentAsXml(content, companyId);
+
+		String title = _getTitleFromContent(content, 3);
+
+		String defaultLocaleId = UpgradeProcessUtil.getDefaultLanguageId(
+			companyId);
+
+		Locale defaultLocale = LocaleUtil.fromLanguageId(defaultLocaleId);
+
+		Map<Locale, String> titleMap = Collections.singletonMap(
+			defaultLocale, title);
+
+		JournalArticle journalArticle =
+			JournalArticleLocalServiceUtil.addArticle(
+				userId, groupId, journalFolderId, titleMap, titleMap,
+				xmlContent, WysiwygConstants.STRUCTURE_KEY,
+				WysiwygConstants.TEMPLATE_KEY, serviceContext);
+
+		return journalArticle;
+	}
+
+	private String _getContentAsXml(String content, long companyId)
+		throws Exception {
+
+		String defaultLocale = UpgradeProcessUtil.getDefaultLanguageId(
+			companyId);
+
+		Document newDocument = SAXReaderUtil.createDocument();
+
+		Element newRootElement = SAXReaderUtil.createElement("root");
+
+		newRootElement.addAttribute("available-locales", defaultLocale);
+		newRootElement.addAttribute("default-locale", defaultLocale);
+
+		newDocument.add(newRootElement);
+
+		Element dynamicElementElement = SAXReaderUtil.createElement(
+			"dynamic-element");
+
+		dynamicElementElement.addAttribute("name", "content");
```

Comment:
You cannot take for granted that the Basic Web Content structure will always have a field named "content" because since 7.0 all structures (including that one) are editable.

You should create your own structure (as suggested in the JIRA ticket)